zerver/actions/presence.py

```diff
@@ -290,6 +290,7 @@ def update_user_presence(
         client,
         status,
     )
-    do_update_user_presence(user_profile, client, log_time, status)
+    if user_profile.presence_enabled:
+        do_update_user_presence(user_profile, client, log_time, status)
     if new_user_input:
```

Comment:
This is wrong - making calling `do_update_user_presence` conditional on `user_profile.presence_enabled` is not relevant for addressing #32605. It's also unnecessary, because `update_user_presence` doesn't get called if the user's presence is disabled to begin with.